src/library_workerfs.js

```diff
@@ -0,0 +1,95 @@
+mergeInto(LibraryManager.library, {
+  $WORKERFS__deps: ['$FS'],
+  $WORKERFS: {
+    DIR_MODE: {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */,
+    FILE_MODE: {{{ cDefine('S_IFREG') }}} | 511 /* 0777 */,
+    mount: function (mount) {
+      assert(ENVIRONMENT_IS_WORKER);
+      var root = WORKERFS.createNode(null, '/', WORKERFS.DIR_MODE, 0);
+      (mount.opts.files || []).forEach(function(file) {
```

Comment:
`File` objects are provided by e.g. `<input type=file>`. They are much handier in use because `File` instance incorporate name, data and timestamp. Also, we can also accept `FileList` here with `Array.prototype.forEach.call(...)`. What do you think?